lib/stdlib/src/zip.erl

```diff
@@ -1534,45 +1534,46 @@ get_file_name_extra(FileNameLen, ExtraLen, B, GPFlag) ->
 get_z_data(?DEFLATED, In0, FileName, CompSize, Input, Output, OpO, Z) ->
     ok = zlib:inflateInit(Z, -?MAX_WBITS),
     Out0 = Output({open, FileName, [write | OpO]}, []),
-    {In1, Out1, UncompSize} = get_z_data_loop(CompSize, 0, In0, Out0, Input, Output, Z),
-    CRC = zlib:crc32(Z),
+    CRC0 = 0,
+    {In1, Out1, UncompSize, CRC} = get_z_data_loop(CompSize, 0, In0, Out0, Input, Output, CRC0, Z),
     ?CATCH zlib:inflateEnd(Z),
     Out2 = Output({close, FileName}, Out1),
     {Out2, In1, CRC, UncompSize};
-get_z_data(?STORED, In0, FileName, CompSize, Input, Output, OpO, Z) ->
+get_z_data(?STORED, In0, FileName, CompSize, Input, Output, OpO, _Z) ->
     Out0 = Output({open, FileName, [write | OpO]}, []),
-    CRC0 = zlib:crc32(Z, <<>>),
-    {In1, Out1, CRC} = copy_data_loop(CompSize, In0, Out0, Input, Output,
-				      CRC0, Z),
+    CRC0 = 0,
+    {In1, Out1, CRC} = copy_data_loop(CompSize, In0, Out0, Input, Output, CRC0),
     Out2 = Output({close, FileName}, Out1),
     {Out2, In1, CRC, CompSize};
 get_z_data(_, _, _, _, _, _, _, _) ->
     throw(bad_file_header).
 
-copy_data_loop(0, In, Out, _Input, _Output, CRC, _Z) ->
+copy_data_loop(0, In, Out, _Input, _Output, CRC) ->
     {In, Out, CRC};
-copy_data_loop(CompSize, In0, Out0, Input, Output, CRC0, Z) ->
+copy_data_loop(CompSize, In0, Out0, Input, Output, CRC0) ->
     N = erlang:min(?READ_BLOCK_SIZE, CompSize),
     case Input({read, N}, In0) of
-	{eof, In1} -> {Out0, In1};
-	{Uncompressed, In1} ->
-	    CRC1 = zlib:crc32(Z, CRC0, Uncompressed),
-	    Out1 = Output({write, Uncompressed}, Out0),
-	    copy_data_loop(CompSize-N, In1, Out1, Input, Output, CRC1, Z)
+        {eof, In1} ->
+            {Out0, In1};
```

Comment:
I wasn't sure if this should return a 3 element tuple to match the other function head. Removing this clause didn't make the zip_SUITE tests fail, I wasn't sure how to trigger this code path either. Any assistance with this would be appreciated.

Reply:
When extracting it will only happen when a file ends before the size given in its header, but that should be caught earlier on when checking the central directory. I think the most reasonable thing to do is to ignore it and crash.

When compressing (`put_z_data_loop`) it will happen if one of the files shrink before we're done reading them, but it seems rather benign there.

Reply:
Thank you, I’ll be happy to send another PR that raises when extracting.